sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2017,6 +2017,8 @@ cl_int ExecCGCommand::enqueueImp() {
       } else
         KnownProgram = false;
     } else {
+      // FIXME should validate if any use-case leads here as in such a case
+      // Kernel handle remains nullptr
```

Comment:
Not sure I understand this comment. Could you please elaborate?

Reply:
There are several scenarios:
 1. Kernel bundle is used
 2. No kernel bundle used but `ExecKernel->MSyclKernel` is non-null
 3. `MSyclKernel` is null

Scenarios 1 and 2 assign some valid value to `Kernel` (which is of type `pi_kernel`). The latter scenario doesn't do this due no valid place to get `pi_kernel` from. Anyway, the `Kernel` variable is used later to set arguments and launch it. The latter scenario will pass nullptr. Hence is the comment.

Reply:
In the third case a kernel is taken from the program manager by calling `detail::ProgramManager::getInstance().getOrCreateKernel`. Am I missing something?